devito/passes/iet/languages/C.py

```diff
@@ -5,12 +5,40 @@
 from devito.passes.iet.definitions import DataManager
 from devito.passes.iet.orchestration import Orchestrator
 from devito.passes.iet.langbase import LangBB
+from devito.passes.iet.languages.utils import _atomic_add_split
 from devito.symbolics import c_complex, c_double_complex
+from devito.symbolics.extended_sympy import UnaryOp
 from devito.tools import dtype_to_cstr
 
 __all__ = ['CBB', 'CDataManager', 'COrchestrator']
 
 
+class RealExt(UnaryOp):
+
+    _op = '__real__ '
+
+
+class ImagExt(UnaryOp):
+
+    _op = '__imag__ '
+
+
+def atomic_add(i, pragmas, split=False):
+    # Base case, real reduction
+    if not split:
+        return i._rebuild(pragmas=pragmas)
+
+    # Complex reduction, split using a temp pointer
+    # Transforms lhs += rhs into
+    # {
+    #   pragmas
+    #   __real__ lhs += __real__ rhs;
```

Comment:
I guess `__real__ lhs` can be assigned to, whilst `creal(lhs)` returns a new value? We currently use `crealf` by default right (at least we do [here](https://github.com/devitocodes/devito/blob/288df4e2d35c2260ee218859b857bb1d2eab8076/tests/test_symbolics.py#L1148))?